ext/duckdb/logical_type.c

```diff
@@ -73,6 +74,26 @@ static VALUE duckdb_logical_type_scale(VALUE self) {
     return INT2FIX(duckdb_decimal_scale(ctx->logical_type));
 }
 
+static VALUE duckdb_logical_type_child_type(VALUE self) {
```

Comment:
I think I will support the following method by using this `duckdb_logical_type_child_type` too.
- [duckdb_array_type_array_size](https://duckdb.org/docs/api/c/api.html#duckdb_array_type_array_size)